aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -1191,7 +1194,7 @@ Tensor view_nested(const Tensor& self, IntArrayRef proposed_shape) {
 Tensor values_nested(const Tensor& self) {
   TORCH_INTERNAL_ASSERT(self.is_nested(), "Can only create a buffer from Nested Tensor");
   auto* nt_self = get_nested_tensor_impl(self);
-  return nt_self->get_buffer();
+  return nt_self->get_unsafe_storage_as_tensor();
```

Comment:
Why not `get_buffer`?

Reply:
So the only place where values is used is in 'unbind' and the NTs are very often not contiguous that get passed into this function takes into account the sizes, strides, and offsets. This brings up the question, does the backward work for .values(). I think yes since it is metadata changes and not actually manipulating data.

Reply:
But then instead of using `values` within the implementation of `unbind`, use `get_unsafe_storage_as_tensor` directly. `values` is also a public, Python facing API. I don't think we want users from Python be exposed to this level of detail unconditionally?

Reply:
What does values do for other tensors, like sparse ones? The .values call is nice because it lets autograd flow through and splits unbind into multiple nodes. I know that we can write an explicit derivative but localizing the interface between regular tensors and NT conversions for autograd is still a nice property that implementing values in this way and re-using it provides.

Reply:
For sparse Tensors `values` contains the materialized values. Sparse tensors compresses some or all zeros and so values contains those elements that weren't compressed (which are either non-zero values or elements with positions that don't fit the mold of a particular layout).

Reply:
I think for NT the public `values` method can be useful for quick prototyping of kernels, but I think it's risky to use it for all the subtle cases as well. cc @albanD

Reply:
Agreed that values should be safe as well! We can move unbind to use another underlying function if needed (narrow(0, i, i+size) for example) and make sure we implement this on in a nice differentiable way.
I don't think this needs to block this PR though.

Reply:
For `unbind` we could also just extend [`select_nested`](https://github.com/pytorch/pytorch/blob/dcc3ae98b7278d9d85be853bfcd070b2a081003f/aten/src/ATen/native/nested/NestedTensorMath.cpp#L673-L692) to work on non-zero dim. For that we'll also need to [extend the derivative](https://github.com/pytorch/pytorch/blob/1cad744694d7feb7c55e5f4ff4a6ae749686bfb5/aten/src/ATen/native/nested/NestedTensorBackward.cpp#L157-L171).